tests/integration/test_astype.py

```diff
@@ -39,8 +61,30 @@ def test_basic(src_dtype, dst_dtype):
     assert np.array_equal(out_num, out_np)
 
 
-@pytest.mark.parametrize("src_dtype", ("F", "D"), ids=to_dtype)
+@pytest.mark.parametrize("src_dtype", ALL_BUT_COMPLEX, ids=to_dtype)
 @pytest.mark.parametrize("dst_dtype", ALL_TYPES, ids=to_dtype)
+@pytest.mark.parametrize("order", ORDER, ids=str)
+def test_order(src_dtype, dst_dtype, order):
+    in_np = np.array(TEST_VECTOR, dtype=src_dtype)
+    in_num = num.array(in_np)
+
+    out_np = in_np.astype(dst_dtype, order=order)
+    out_num = in_num.astype(dst_dtype, order=order)
+
+    assert np.array_equal(out_num, out_np)
+
+
+@pytest.mark.parametrize("src_dtype", ALL_BUT_COMPLEX[1:5], ids=to_dtype)
+@pytest.mark.parametrize("dst_dtype", ALL_BUT_COMPLEX[5:8], ids=to_dtype)
```

Comment:
This slicing seems fragile, if anyone happens to change the contents of `ALL_BUT_COMPLEX` without realizing it is used like this, this test could fail in subtle ways. I would make a new dedicated named lists, or just past list literals of the types here.

Reply:
Redefined this variable